src/gui/qgsgradientcolorrampdialog.cpp

```diff
@@ -66,11 +69,18 @@ QgsGradientColorRampDialog::QgsGradientColorRampDialog( const QgsGradientColorRa
 
   connect( mStopColorSpec, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, [ = ]
   {
-    mStopDirection->setEnabled( static_cast< QColor::Spec>( mStopColorSpec->currentData().toInt() ) != QColor::Spec::Rgb );
+    const QColor::Spec colorSpec = static_cast< QColor::Spec>( mStopColorSpec->currentData().toInt() );
+    mStopDirection->setEnabled( colorSpec != QColor::Spec::Rgb );
```

Comment:
Direction should be disabled for cmyk too, right? Let's use a switch here on colorSpec so that we explicitly handle all cases.

Reply:
Right! done